File: sale-process/sales-app/src/app/interfaces/venta.ts
import { Cliente } from './cliente';
import { Usuario } from './usuario';
import { DetalleVenta } from './detalle-venta';

export interface Venta {
  id: number;
  tipoComprobante: string; 
  cliente: Cliente;  
  usuario: Usuario;  
  fechaEmision: string;  
  subtotal: number;
  igv: number;
  total: number;
  estado: boolean;
  detalleVentas: DetalleVenta[];  
}

File: sale-process/sales/src/main/java/sales/system/sales/model/Venta.java
package sales.system.sales.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;
import jakarta.persistence.*;

import java.util.Date;
import java.util.List;

@Data
@Entity
public class Venta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    private Date fechaEmision;
    private Double subtotal;
    private String tipoComprobante;
    private Double igv;
    private Double total;

    private Boolean estado = true;

    // Usamos @JsonManagedReference en el lado "padre" de la relación
    @OneToMany(mappedBy = "venta")
    @JsonManagedReference
    private List<DetalleVenta> detalleVentas;
}
